core/store/src/lib.rs

```diff
@@ -189,6 +190,18 @@ impl Store {
         }
     }
 
+    pub fn get_or_err(&self, column: DBCol, key: &[u8]) -> Result<Vec<u8>, Error> {
```

Comment:
I’m not sure.  It doesn’t feel right that this returns a chain Error.  Chain is an abstraction layer above Store.  I think I’d rather this returned io::Result like the other functions.

Another approach is to introduce

```
trait StoreExt {
    pub get_or_err(...) ...;
    pub get_ser_or_err(...) ...;
}
```

in the module that you need it in.

Reply:
I'd maybe just ask the caller to call `option_to_not_found` manually?